File: Advanced Business Development With dotNET/Models/Moto.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MottuFlowApi.Models
{
    [Table("moto")]
    public class Moto
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("id_moto")]
        public int IdMoto { get; set; }

        [Required]
        [MaxLength(10)]
        [Column("placa")]
        public string Placa { get; set; } = string.Empty;

        [Required]
        [MaxLength(50)]
        [Column("modelo")]
        public string Modelo { get; set; } = string.Empty;

        [Required]
        [MaxLength(50)]
        [Column("fabricante")]
        public string Fabricante { get; set; } = string.Empty;

        [Required]
        [Column("ano")]
        public int Ano { get; set; }

        [Required]
        [Column("id_patio")]
        public int IdPatio { get; set; }

        [Required]
        [MaxLength(100)]
        [Column("localizacao_atual")]
        public string LocalizacaoAtual { get; set; } = string.Empty;

        // Navigation
        public Patio? Patio { get; set; }
        public List<ArucoTag> ArucoTags { get; set; } = new List<ArucoTag>();
        public List<Localidade> Localidades { get; set; } = new List<Localidade>();
        public List<RegistroStatus> RegistrosStatus { get; set; } = new List<RegistroStatus>();
    }
}

File: Compliance Quality Assurance and Tests/MottuFlow/src/main/java/com/sprint/MottuFlow/domain/moto/Moto.java
package com.sprint.MottuFlow.domain.moto;

import com.sprint.MottuFlow.domain.arucotag.ArucoTag;
import com.sprint.MottuFlow.domain.localidade.Localidade;
import com.sprint.MottuFlow.domain.patio.Patio;
import com.sprint.MottuFlow.domain.status.Status;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Table(name = "moto")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Moto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idMoto;

    @Column(nullable = false, length = 10)
    private String placa;

    @Column(nullable = false, length = 50)
    private String modelo;

    @Column(nullable = false, length = 50)
    private String fabricante;

    @Column(nullable = false)
    private int ano;

    @ManyToOne
    @JoinColumn(name = "id_patio", nullable = false)
    private Patio patio;

    @Column(nullable = false, length = 100)
    private String localizacaoAtual;

    @OneToMany(mappedBy = "moto", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ArucoTag> arucoTags;

    @OneToMany(mappedBy = "moto", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Status> estados;

    @OneToMany(mappedBy = "moto", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Localidade> localidades;

	public Long getIdMoto() {
		return idMoto;
	}

	public void setIdMoto(Long idMoto) {
		this.idMoto = idMoto;
	}

	public String getPlaca() {
		return placa;
	}

	public void setPlaca(String placa) {
		this.placa = placa;
	}

	public String getModelo() {
		return modelo;
	}

	public void setModelo(String modelo) {
		this.modelo = modelo;
	}

	public String getFabricante() {
		return fabricante;
	}

	public void setFabricante(String fabricante) {
		this.fabricante = fabricante;
	}

	public int getAno() {
		return ano;
	}

	public void setAno(int ano) {
		this.ano = ano;
	}

	public Patio getPatio() {
		return patio;
	}

	public void setPatio(Patio patio) {
		this.patio = patio;
	}

	public String getLocalizacaoAtual() {
		return localizacaoAtual;
	}

	public void setLocalizacaoAtual(String localizacaoAtual) {
		this.localizacaoAtual = localizacaoAtual;
	}

	public List<ArucoTag> getArucoTags() {
		return arucoTags;
	}

	public void setArucoTags(List<ArucoTag> arucoTags) {
		this.arucoTags = arucoTags;
	}

	public List<Status> getEstados() {
		return estados;
	}

	public void setEstados( List<Status> status ) {
		this.estados = estados;
	}

	public List<Localidade> getLocalidades() {
		return localidades;
	}

	public void setLocalidades(List<Localidade> localidades) {
		this.localidades = localidades;
	}
}